### PR TITLE
feat: add domain_whitelist.kv3

### PR DESCRIPTION
**Pull request by craftablescience**
~~Ties in to internal MR adding a whitelist to `AsyncWebRequest`, fine to merge *after that MR is in*~~
Internal MR has been merged, this should be merged before your next release

### Checks

-   [x] **I have thoroughly tested all of the code I have modified/added/removed to ensure something else did not break**
-   [x] I have followed [semantic commit messages](https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716) e.g. `feat: Add foo`, `chore: Update bar`, etc...
-   [x] My branch has a clear history of changes that can be easy to follow when being reviewed commit-by-commit
-   [x] My branch is functionally complete; the only changes to be done will be those potentially requested in code review
-   [x] All changes requested in review have been `fixup`ed into my original commits.
-   [x] Fully tokenized all my strings (no hardcoded English strings!!) and supplied [bulk JSON strings](https://docs.momentum-mod.org/guide/localization/#bulk-adding-terms) below

### CODE REVIEW

File: domain_whitelist.kv3 [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Bypass restricting web requests to this list of domains by adding the "-unrestrictedwebrequests" launch argument
+{
+	domains =
+	[
+		"steampowered.com",
+		"steamcommunity.com",
+		"valvesoftware.com",
```
Comment (tsa96) [inline domain_whitelist.kv3:7]: Since this is per-game (very happy it is), don't see the point of us supporting these domains

**Review by tsa96 — APPROVED** [review]

**Review by tsa96 — CHANGES_REQUESTED** [review]
We don't need to Steam URLs IMO

File: domain_whitelist.kv3 (hunk 1 above)
Comment (craftablescience) [reply domain_whitelist.kv3]: Removed these domains

**Review by tsa96 — APPROVED** [review]

**Review by Gocnak — APPROVED** [review]
We do all API requests in c++ code so this is fine

**Pull request merged**